### DISCUSSION TITLE
Azure Container Registry

### DISCUSSION THREAD

**Discussion by damianopezzotti** [artifacthub/hub#1789] [General]
Hi,
I have some Helm Charts deployed on Azure Container Registry as OCI.
Is it possible to connect ArtifactHub on this repository?
I tried to add a repository specifying the following parameters:

- name: myrepo
- display name: myrepo
- URL oci://myrepo.azurecr.io
- username
- password

but I obtain the following error:

`error getting repository remote digest: GET https://auth.docker.io/token?scope=repository%3Alibrary%2Fxxx.azurecr.io%3Apull&service=registry.docker.io: unexpected status code 401 Unauthorized: {"details":"incorrect username or password"}
`

Any Idea?
Thanks

**Comment by tegioz** [comment]
Hi @damianopezzotti 👋 

It looks like the namespace and repository parts are missing in the repository URL (the `helm/hello-world` part in this [example](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-helm-repos#install-helm-chart)). Without that, `go-containerregistry`, a library used by Artifact Hub, [considers the registry is the Docker Hub](https://github.com/google/go-containerregistry/blob/890d5b362eb849db112bb36543014e4e5e57ef8d/pkg/name/repository.go#L81-L87) and the rest is the repository part. Please note the credentials set up have been tried against the Docker Hub, hence the error (just in case you want to revoke them).

Hope this helps.

**Comment by damianopezzotti** [comment]
Thanks, it works!
Is it possible to scan multiple charts in a single repo?
I have my helm charts inside oci://myrepo.azurecr.io/helm/ (e.g. oci://myrepo.azurecr.io/helm/sample1, oci://myrepo.azurecr.io/helm/sample2, etc...)

**Comment by tegioz** [comment]
Awesome, no worries 👍 

No, that's not possible at the moment. `helm` in this case would actually be a namespace, not a repo. Unfortunately you'll need to add each of them separately. If you have a lot of them maybe you'd like to take a look at the [API](https://artifacthub.io/docs/api/) to automate the process.

**Reply by damianopezzotti** [reply]
Thanks for your answer

Regards
Damiano
